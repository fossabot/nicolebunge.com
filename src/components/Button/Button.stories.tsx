import Button from '.';

export default {
  component: Button,
};

export const Default = {
  args: {
    children: 'Button',
    to: '/',
  },
};
