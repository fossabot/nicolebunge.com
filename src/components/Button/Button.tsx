import clsx from 'clsx';
import { Link } from 'gatsby';
import React, { ReactNode } from 'react';
import * as styles from './Button.module.css';

export interface ButtonProps {
  className?: string;
  children: ReactNode;
  to: string;
}

function Button(props: ButtonProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <Link className={clsx(className, styles.button)} {...otherProps} />;
}

export default Button;
