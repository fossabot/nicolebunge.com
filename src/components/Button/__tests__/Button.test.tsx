/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Button from '..';

describe('Button component', () => {
  it('renders correctly', () => {
    const { container } = render(<Button to="/">Button</Button>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
