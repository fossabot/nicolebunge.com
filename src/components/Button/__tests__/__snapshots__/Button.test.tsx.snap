// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<a
  class="button"
  href="/"
>
  Button
</a>
`;
